src/sentry/dynamic_sampling/prioritise_projects.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.sentry_metrics.indexer.strings import TRANSACTION_METRICS_NAMES
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 1000
+
+
+def fetch_projects_with_total_volumes() -> Mapping[int, Sequence[int]]:
+    """
+    This function fetch with pagination orgs and projects with count per root project
+    """
+    aggregated_projects = defaultdict(list)
+    start_time = time.time()
+    offset = 0
+    while (time.time() - start_time) < MAX_SECONDS:
+        query = (
+            Query(
+                match=Entity(EntityKey.GenericOrgMetricsCounters.value),
+                select=[
+                    Function("sum", [Column("value")], "root_count_value"),
+                    Column("org_id"),
+                    Column("project_id"),
+                ],
+                groupby=[Column("org_id"), Column("project_id")],
+                where=[
+                    Condition(Column("timestamp"), Op.GTE, datetime.utcnow() - timedelta(hours=6)),
+                    Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                    Condition(
+                        Column("metric_id"),
+                        Op.EQ,
+                        TRANSACTION_METRICS_NAMES[TransactionMRI.COUNT_PER_ROOT_PROJECT.value],
+                    ),
+                ],
+                granularity=Granularity(3600),
+                orderby=[
+                    OrderBy(Column("org_id"), Direction.ASC),
+                    OrderBy(Column("project_id"), Direction.ASC),
+                ],
```

Comment:
Do we need `order_by` for correct pagination, i.e. offsets between query are variable without an order ?
If the return from snuba is stable then we shouldn't care about the order.

Reply:
we inherit the guarantees from clickhouse here. orderby is necessary for deterministic order. I believe there are also no concerns wrt its performance since the entire table is sorted by those two fields